src/utility/Albany_ThyraCrsMatrixFactory.cpp

```diff
@@ -28,34 +25,35 @@ struct ThyraCrsMatrixFactory::Impl {
 
 ThyraCrsMatrixFactory::
 ThyraCrsMatrixFactory (const Teuchos::RCP<const Thyra_VectorSpace> domain_vs,
-                    const Teuchos::RCP<const Thyra_VectorSpace> range_vs,
-                    const int nonzeros_per_row,
-                    const bool static_profile)
+                       const Teuchos::RCP<const Thyra_VectorSpace> range_vs,
+                       const int /*nonzeros_per_row*/)
  : m_graph(new Impl())
  , m_domain_vs(domain_vs)
  , m_range_vs(range_vs)
  , m_filled (false)
- , m_static_profile (static_profile) 
 {
   auto bt = Albany::build_type();
```

Comment:
This is orthogonal to the PR, but it came to my mind: should we check the type of the two vs rather than the build type? There's no reason to rely on the build type rather than the nature of the two vs, especially since the two vs will later be casted down to E or T maps...

Reply:
Are you are saying that we should try casting them into Epetra/Tpetra maps  and check whether they are null or not?